tensorflow_datasets/image/cycle_gan.py

```diff
@@ -0,0 +1,213 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""CycleGAN dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+import numpy as np
+import tensorflow as tf
+
+from tensorflow_datasets.core import api_utils
+import tensorflow_datasets.public_api as tfds
+
+
+_DL_URL = "https://people.eecs.berkeley.edu/~taesung_park/CycleGAN/datasets/"
+_DL_URLS = {
+    "ae_photos": _DL_URL + "ae_photos.zip",
+    "apple2orange": _DL_URL + "apple2orange.zip",
+    "summer2winter_yosemite": _DL_URL + "summer2winter_yosemite.zip",
+    "horse2zebra": _DL_URL + "horse2zebra.zip",
+    "monet2photo": _DL_URL + "monet2photo.zip",
+    "cezanne2photo": _DL_URL + "cezanne2photo.zip",
+    "ukiyoe2photo": _DL_URL + "ukiyoe2photo.zip",
+    "vangogh2photo": _DL_URL + "vangogh2photo.zip",
+    "maps": _DL_URL + "maps.zip",
+    "cityscapes": _DL_URL + "cityscapes.zip",
+    "facades": _DL_URL + "facades.zip",
+    "iphone2dslr_flower": _DL_URL + "iphone2dslr_flower.zip",
+}
+
+_DATA_OPTIONS = ["ae_photos", "apple2orange", "summer2winter_yosemite", "horse2zebra", "monet2photo", "cezanne2photo", "ukiyoe2photo", "vangogh2photo", "maps", "cityscapes", "facades", "iphone2dslr_flower"]
+
+
+class CycleGANConfig(tfds.core.BuilderConfig):
+  """BuilderConfig for CycleGAN.""" 
+  
+  @api_utils.disallow_positional_args
+  def __init__(self, data=None, **kwargs):
+    """Constructs a CycleGANConfig. 
+    Args:
+      data: `str`, one of `_DATA_OPTIONS`.
+      **kwargs: keyword arguments forwarded to super.
+    """
+    if data not in _DATA_OPTIONS:
+      raise ValueError("data must be one of %s" % _DATA_OPTIONS)
+  
+    super(CycleGANConfig, self).__init__(**kwargs)
+    self.data = data
+
+    
+class CycleGAN(tfds.core.GeneratorBasedBuilder):
+  """CycleGAN dataset."""
+  
+  VERSION = tfds.core.Version("2.0.0")
```

Comment:
what's the reason for 2.0?

Reply:
What should I use here ? Should I use <code>tfds.core.Version('0.1.0')</code>. Please suggest. I see something as <code>MAJOR.MINOR.PATCH</code>.

Reply:
We usually start with 0.1.0 - if dataset is under development and then eventually 1.0.0

Using 2.0 is not an error per se - but it suggests that we had an earlier version

Reply:
Yes, I changed it to "0.1.0". Thanks.